packages/sdk-middleware-http/src/http.js

```diff
@@ -41,68 +47,82 @@ export default function createHttpMiddleware (
+            return
+          }
+
+          // Server responded with an error. Try to parse it as JSON, then
+          // return a proper error type with all necessary meta information.
+          res.text()
+          .then((text: any) => {
+            // Try to parse the error response as JSON
+            let parsed
+            try {
+              parsed = JSON.parse(text)
+            } catch (error) {
+              /* noop */
+            }
+
+            const error: HttpErrorType = createError({
+              statusCode: res.status,
+              originalRequest: request,
+              headers: parseHeaders(res.headers),
+              ...(parsed
+                ? { message: parsed.message, body: parsed }
+                : {}
+              ),
+            })
+            // Let the final resolver to reject the promise
+            const parsedResponse = {
               ...response,
-              body: result,
+              error,
               statusCode: res.status,
             }
-            if (options.includeResponseHeaders)
-              parsedResponse.headers = parseHeaders(res.headers)
-            if (options.includeOriginalRequest) {
-              parsedResponse.request = {
-                ...requestObj,
-                headers: parseHeaders(requestObj.headers),
-              }
-              if (options.maskSensitiveHeaderData)
-                parsedResponse.request.headers.authorization = 'Bearer ********'
-            }
             next(request, parsedResponse)
           })
-          return
-        }
-
-        // Server responded with an error. Try to parse it as JSON, then return
-        // a proper error type with all necessary meta information.
-        res.text()
-        .then((text: any) => {
-          // Try to parse the error response as JSON
-          let parsed
-          try {
-            parsed = JSON.parse(text)
-          } catch (error) {
-            /* noop */
-          }
-
-          const error: HttpErrorType = createError({
-            statusCode: res.status,
-            originalRequest: request,
-            headers: parseHeaders(res.headers),
-            ...(parsed
-              ? { message: parsed.message, body: parsed }
-              : {}
-            ),
-          })
-          // Let the final resolver to reject the promise
-          const parsedResponse = {
-            ...response,
-            error,
-            statusCode: res.status,
-          }
-          next(request, parsedResponse)
-        })
-      },
-      // We know that this is a "network" error thrown by the `fetch` library
-      (e: Error) => {
-        const error = new NetworkError(e.message, { originalRequest: request })
-        next(request, { ...response, error, statusCode: 0 })
-      },
-    )
+        },
+        // We know that this is a "network" error thrown by the `fetch` library
+        (e: Error) => {
+          if (enableRetry)
+            if (retryCount < maxRetries) {
```

Comment:
The idea is rather sdk does not log anything but let the caller log it if required. Sdk should just throw custom error with custom message. Custom error should always supply root error/message